packages/flutter_tools/lib/src/hot.dart

```diff
@@ -151,7 +109,18 @@ class HotRunner extends ResidentRunner {
         new DartDependencySetBuilder(
               _mainPath, _projectRootPath, _packagesFilePath);
     try {
-      _dartDependencies = dartDependencySetBuilder.build();
+      Set<String> dependencies = dartDependencySetBuilder.build();
+      _dartDependencies = new Set<String>();
+      for (String path in dependencies) {
+        // We need to tweak package: uris so that they reflect their devFS
+        // location.
+        if (path.startsWith('package:')) {
+          // Swap out package: for packages/ because we place all package
+          // sources under packages/.
+          path = path.replaceFirst('package:', 'packages/');
```

Comment:
Because the two consumers of DartDependencySetBuilder want different transformations of package: uris.